nimx/button.nim

```diff
@@ -95,11 +95,20 @@ proc drawTitle(b: Button, xOffset: Coord) =
             else:
                 blackColor()
 
-        let font = systemFont()
-        var titleRect = b.bounds
-        var pt = centerInRect(font.sizeOfString(b.title), titleRect)
+        # truncate title to bounds if required
+        var
+            titleRect = b.bounds
+            title = b.title
+        let 
+            font = systemFont()
+            sz = font.sizeOfString(title)
+        # reduce by size of checkbox/radiobox
+        titleRect.size.width -= titleRect.size.height
+        if sz.width > titleRect.size.width:
+            title.setLen(int(title.len.float * titleRect.size.width / sz.width))
+        var pt = centerInRect(sz, titleRect)
         if pt.x < xOffset: pt.x = xOffset
```

Comment:
I might be wrong but you'll have to account for `xOffset` before trimming. Otherwise trimming done above might not be sufficient and the text will still escape the bounds by up to `xOffset`.